### PR TITLE
Xframe and preferred login changes

### PR DESCRIPTION
**Pull request by rashmi43**

### CODE REVIEW

File: pkg/controller/authentication/ingress.go [hunk 1]
```diff
@@ -804,6 +810,13 @@ func platformOidcIngress(instance *operatorv1alpha1.Authentication, scheme *runt
 			},
 		},
 	}
+	if ingressUpdateRequired {
+		err = r.client.Update(context.TODO(), newIngress)
```
Comment (posriniv) [inline pkg/controller/authentication/ingress.go:814]: where are you setting this variable? @rashmi43

File: pkg/controller/authentication/containers.go [hunk 2]
```diff
@@ -654,7 +654,7 @@ func buildIdentityProviderContainer(instance *operatorv1alpha1.Authentication, i
 		},
 	}
 
-	idpEnvVarList := []string{"NODE_ENV", "LOG_LEVEL_IDPROVIDER", "LOG_LEVEL_MW", "PROVIDER_ISSUER_URL", "IDTOKEN_LIFETIME", "ROKS_ENABLED", "ROKS_URL", "OS_TOKEN_LENGTH", "LIBERTY_TOKEN_LENGTH",
+	idpEnvVarList := []string{"NODE_ENV", "LOG_LEVEL_IDPROVIDER", "LOG_LEVEL_MW", "PROVIDER_ISSUER_URL", "PREFERRED_LOGIN", "IDTOKEN_LIFETIME", "ROKS_ENABLED", "ROKS_URL", "OS_TOKEN_LENGTH", "LIBERTY_TOKEN_LENGTH",
 		"IDENTITY_PROVIDER_URL", "BASE_AUTH_URL", "BASE_OIDC_URL", "SCOPE_CLAIM", "OIDC_ISSUER_URL", "HTTP_ONLY"}
```
Comment (posriniv) [inline pkg/controller/authentication/containers.go:658]: @rashmi43 Don't you need XFRAME_DOMAIN here?

File: pkg/controller/authentication/configmap.go [hunk 3]
```diff
@@ -233,6 +240,7 @@ func authIdpConfigMap(instance *operatorv1alpha1.Authentication, scheme *runtime
 			"SCOPE_CLAIM":                        instance.Spec.Config.ScopeClaim,
 			"BOOTSTRAP_USERID":                   instance.Spec.Config.BootstrapUserId,
 			"PROVIDER_ISSUER_URL":                instance.Spec.Config.ProviderIssuerURL,
```
Comment (posriniv) [inline pkg/controller/authentication/configmap.go:242]: @rashmi43 Don't you need XFRAME_DOMAIN here?

File: pkg/controller/authentication/ingress.go [hunk 4]
```diff
@@ -804,6 +810,13 @@ func platformOidcIngress(instance *operatorv1alpha1.Authentication, scheme *runt
 			},
 		},
 	}
+	if ingressUpdateRequired {
```
Comment (posriniv) [inline pkg/controller/authentication/ingress.go:813]: Where are you setting this variable @rashmi43

**Review by posriniv — CHANGES_REQUESTED** [review]

File: pkg/controller/authentication/containers.go (hunk 2 above)
Comment (rashmi43) [reply pkg/controller/authentication/containers.go:658]: No I do not

File: pkg/controller/authentication/ingress.go (hunk 1 above)
Comment (rashmi43) [reply pkg/controller/authentication/ingress.go:814]: Wanted to ask you that

File: pkg/controller/authentication/ingress.go (hunk 4 above)
Comment (posriniv) [inline pkg/controller/authentication/ingress.go:813]: @rashmi43 you can remove this whole block, we won't have a configmap kind of scenario here for the ingress.

**Review by posriniv — CHANGES_REQUESTED** [review]

File: pkg/controller/authentication/configmap.go [hunk 5]
```diff
@@ -157,6 +157,13 @@ func (r *ReconcileAuthentication) handleConfigMap(instance *operatorv1alpha1.Aut
 					currentConfigMap.Data["PROVIDER_ISSUER_URL"] = newConfigMap.Data["PROVIDER_ISSUER_URL"]
 					cmUpdateRequired = true
 				}
+				if _, keyExists := currentConfigMap.Data["PREFERRED_LOGIN"]; !keyExists {
+					reqLogger.Info("Updating an existing Configmap", "Configmap.Namespace", currentConfigMap.Namespace, "ConfigMap.Name", currentConfigMap.Name)
+					newConfigMap = functionList[index](instance, r.scheme)
+					currentConfigMap.Data["PREFERRED_LOGIN"] = newConfigMap.Data["PREFERRED_LOGIN"]
+					currentConfigMap.Data["XFRAME_DOMAIN"] = newConfigMap.Data["XFRAME_DOMAIN"]
+					cmUpdateRequired = true
```
Comment (posriniv) [inline pkg/controller/authentication/configmap.go:165]: @rashmi43 Remove XFRAME_DOMAIN here

**Review by posriniv — CHANGES_REQUESTED** [review]

File: pkg/controller/authentication/configmap.go [hunk 6]
```diff
@@ -233,6 +239,7 @@ func authIdpConfigMap(instance *operatorv1alpha1.Authentication, scheme *runtime
 			"SCOPE_CLAIM":                        instance.Spec.Config.ScopeClaim,
 			"BOOTSTRAP_USERID":                   instance.Spec.Config.BootstrapUserId,
 			"PROVIDER_ISSUER_URL":                instance.Spec.Config.ProviderIssuerURL,
+			"PREFERRED_LOGIN    ":                instance.Spec.Config.PreferredLogin,
 			"LIBERTY_TOKEN_LENGTH":               "1024",
```
Comment (posriniv) [inline pkg/controller/authentication/configmap.go:243]: @rashmi43 why is there additional space after PREFERRED_LOGIN? Please remove it

**Review by posriniv — CHANGES_REQUESTED** [review]

**Review by posriniv — APPROVED** [review]
lgtm

**Pull request merged**